offchain/authority-claimer/src/checker.rs

```diff
@@ -146,7 +143,10 @@ impl DuplicateChecker for DefaultDuplicateChecker {
 }
 
 impl DefaultDuplicateChecker {
-    async fn update_claims(&mut self) -> Result<(), DuplicateCheckerError> {
+    async fn update_claims(
+        &mut self,
+        dapp_address: Address,
+    ) -> Result<(), DuplicateCheckerError> {
```

Comment:
This will not work. This function needs to read and store the claims for all dapps.
Otherwise, you would need a `self.next_block_to_read` for each dapp, which might be inefficient.